web/html/src/manager/virtualization/guests/edit/guests-edit.js

```diff
@@ -7,56 +7,75 @@ const {Panel} = require("components/panel");
 const MessagesUtils = require("components/messages").Utils;
 const {GuestProperties} = require('../guest-properties');
 const {VirtualizationGuestActionApi} = require('../virtualization-guest-action-api');
+const {VirtualizationGuestDefinitionApi} = require('../virtualization-guest-definition-api');
 
 
 class GuestsEdit extends React.Component {
 
   constructor(props) {
     super(props);
-    this.initialGuestMemory = props.guest.memory / 1024,
-    this.initialGuestCpu = props.guest.vcpus,
+  }
 
-    this.getInitialModel = () => {
+  getModelFromDefinition(definition) {
+    if (definition) {
+      return {
+        memory: definition.maxMemory / 1024,
+        vcpu: definition.vcpu.max,
+      }
+    } else {
       return {
-        memory: this.initialGuestMemory,
-        vcpu: this.initialGuestCpu,
+        memory: 1024,
+        vcpu: 1,
       }
-    };
+    }
   }
 
   render() {
     return (
-      <VirtualizationGuestActionApi
-        hostid={this.props.host.id}
-        bounce={`/rhn/manager/systems/details/virtualization/guests/${this.props.host.id}`}>
+      <VirtualizationGuestDefinitionApi hostid={this.props.host.id}
+        guestUuid={this.props.guestUuid}>
         {
           ({
-            onAction,
-            messages,
+            definition,
+            definitionMessages
           }) => {
-            const onSubmit = (properties) => {
-              properties.memory = properties.memory * 1024;
-              return onAction('update', [this.props.guest.uuid], properties);
-            };
             return (
-              <Panel title={this.props.guest.name} icon="fa spacewalk-icon-virtual-guest">
-                <GuestProperties
-                  host={this.props.host}
-                  submitText={t("Update")}
-                  submit={onSubmit}
-                  messages={messages}
-                  getInitialModel={this.getInitialModel}/>
-              </Panel>
-            );
+              <VirtualizationGuestActionApi
+                hostid={this.props.host.id}
+                bounce={`/rhn/manager/systems/details/virtualization/guests/${this.props.host.id}`}>
+                {
+                  ({
+                    onAction,
+                    actionMessages,
+                  }) => {
+                    const onSubmit = (properties) => {
+                      properties.memory = properties.memory * 1024;
```

Comment:
uh modifying directly the input here isn't pretty! But I bet you will need to correct this after activating the linting :D

Reply:
I just moved the linting fixes PR before this one... so this is fixed, yeah